armi/apps.py

```diff
@@ -260,13 +260,38 @@ def registerUserPlugins(self, pluginPaths):
         restrict their flexibility and power as compared to the usual ArmiPlugins.
         """
         for pluginPath in pluginPaths:
+            if self._isPluginRegistered(pluginPath):
```

Comment:
The thing that convinces me that this PR is okay is that this is only implemented for `UserPlugin`s, not ALL `Plugin`s.